include/triton/Dialect/TritonGPU/IR/TritonGPUDialect.td

```diff
@@ -23,12 +23,14 @@ def TritonGPU_Dialect : Dialect {
     void registerTypes();
 
     LinearLayout toLinearLayout(ArrayRef<int64_t> shape, Attribute layout);
```

Comment:
We could also make this method return a `const LinearLayout&` pointing to the element of the cache. I did not do that yet, as I think that we are not going to need to create so many raw LLs. Most of them are going to be wrapped in LinearEncodings.

If we ever profile compile times this will probably be one of the first optimisations we should look into doing tho